From: Nico Berlee <nico.berlee@on2it.net>

This patch addresses an issue where Kernel Address Space Layout Randomization
(KASLR) fails on RK3588 systems due to insufficient entropy during the boot
process. The root cause is that the hardware Random Number Generator (RNG)
within the kernel is not initialized early enough to provide the necessary
seed for KASLR.

To resolve this, the patch enables the Rockchip RNG driver in the U-Boot
bootloader. This approach allows the system to gather sufficient entropy
early in the boot process. The seed generated by the RNG is then passed to
the kernel using the EFI_RNG_PROTOCOL, ensuring that KASLR can function
correctly.

Signed-off-by: Nico Berlee <nico.berlee@on2it.net>

diff --git a/configs/turing-rk1-rk3588_defconfig b/configs/turing-rk1-rk3588_defconfig
index 289f2da..44c1155 100644
--- a/configs/turing-rk1-rk3588_defconfig
+++ b/configs/turing-rk1-rk3588_defconfig
@@ -133,3 +133,6 @@ CONFIG_USB_GADGET_PRODUCT_NUM=0x350b
 CONFIG_USB_GADGET_DOWNLOAD=y
 CONFIG_USB_FUNCTION_ROCKUSB=y
 CONFIG_ERRNO_STR=y
+CONFIG_EFI_RNG_PROTOCOL=y
+CONFIG_DM_RNG=y
+CONFIG_RNG_ROCKCHIP=y
diff --git a/arch/arm/dts/rk3588s-u-boot.dtsi b/arch/arm/dts/rk3588s-u-boot.dtsi
index c0fd16c..50de26b 100644
--- a/arch/arm/dts/rk3588s-u-boot.dtsi
+++ b/arch/arm/dts/rk3588s-u-boot.dtsi
@@ -120,7 +120,6 @@
 	rng: rng@fe378000 {
 		compatible = "rockchip,trngv1";
 		reg = <0x0 0xfe378000 0x0 0x200>;
-		status = "disabled";
 	};
 
 	usbdp_phy0: phy@fed80000 {
